avm/ptn/ai-platform/baseline/main.bicep

```diff
@@ -0,0 +1,426 @@
+  }
+}
+
+module applicationInsights 'br/public:avm/res/insights/component:0.3.1' = {
+  name: '${uniqueString(deployment().name, location)}-appi'
+  params: {
+    name: applicationInsightsSettings.?name ?? 'appi-${name}'
+    location: location
+    kind: 'web'
+    enableTelemetry: enableTelemetry
+    workspaceResourceId: logAnalyticsWorkspace.id
+    roleAssignments: [
+      {
+        principalId: managedIdentity.properties.principalId
+        roleDefinitionIdOrName: 'Contributor'
+      }
+    ]
+    tags: tags
+  }
+}
+
+module workspaceHub 'br/public:avm/res/machine-learning-services/workspace:0.4.0' = {
+  name: '${uniqueString(deployment().name, location)}-hub'
+  params: {
+    name: workspaceHubSettings.?name ?? 'hub-${name}'
+    sku: 'Standard'
+    location: location
+    enableTelemetry: enableTelemetry
+    kind: 'Hub'
+    associatedApplicationInsightsResourceId: applicationInsights.outputs.resourceId
+    associatedKeyVaultResourceId: keyVault.outputs.resourceId
+    associatedStorageAccountResourceId: storageAccount.outputs.resourceId
+    associatedContainerRegistryResourceId: containerRegistry.outputs.resourceId
+    workspaceHubConfig: {
+      defaultWorkspaceResourceGroup: resourceGroup().id
+    }
+    managedIdentities: {
+      userAssignedResourceIds: [
+        managedIdentity.id
+      ]
+    }
+    primaryUserAssignedIdentity: managedIdentity.id
+    computes: workspaceHubSettings.?computes
+    managedNetworkSettings: {
+      isolationMode: workspaceHubSettings.?networkIsolationMode ?? 'AllowInternetOutbound'
+      outboundRules: workspaceHubSettings.?networkOutboundRules
+    }
+    systemDatastoresAuthMode: 'identity'
+    roleAssignments: [
+      {
+        principalId: managedIdentity.properties.principalId
+        roleDefinitionIdOrName: 'Contributor'
+      }
+    ]
+    tags: tags
+  }
+}
+
+// ============ //
+// Outputs      //
+// ============ //
+
+@description('The name of the resource group the module was deployed to.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The location the module was deployed to.')
+output location string = location
+
+@description('The resource ID of the application insights component.')
+output applicationInsightsResourceId string = applicationInsights.outputs.resourceId
+
+@description('The name of the application insights component.')
+output applicationInsightsName string = applicationInsights.outputs.name
+
+@description('The application ID of the application insights component.')
+output applicationInsightsApplicationId string = applicationInsights.outputs.applicationId
+
+@description('The instrumentation key of the application insights component.')
+output applicationInsightsInstrumentationKey string = applicationInsights.outputs.instrumentationKey
+
+@description('The connection string of the application insights component.')
+output applicationInsightsConnectionString string = applicationInsights.outputs.connectionString
```

Comment:
Couldn't either of those be considered 'confidential'?

Reply:
I guess the answer is no 😄 https://learn.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=dotnet5#is-the-connection-string-a-secret